File: app/components/folio/console/html_auto_format/toggle_component.js
window.FolioConsole = window.FolioConsole || {}
window.FolioConsole.HtmlAutoFormat = window.FolioConsole.HtmlAutoFormat || {}

window.Folio.Stimulus.register('f-c-html-auto-format-toggle', class extends window.Stimulus.Controller {
  static values = {
    enabled: Boolean,
    apiUrl: String,
  }

  enabledValueChanged () {
    window.FolioConsole.HtmlAutoFormat.enabled = this.enabledValue

    window.Folio.Api.apiPost(this.apiUrlValue, { html_auto_format: this.enabledValue })
  }

  booleanToggleInput (e) {
    this.enabledValue = e.detail.checked
  }
})

File: app/components/folio/console/html_auto_format/toggle_component.rb
# frozen_string_literal: true

class Folio::Console::HtmlAutoFormat::ToggleComponent < Folio::Console::ApplicationComponent
  def initialize; end

  def preference_toggle_component
    Folio::Console::CurrentUsers::PreferenceToggleComponent.new(key: "html_auto_format",
                                                                javascript_key: "HtmlAutoFormat",
                                                                label: t(".label"),
                                                                class_name: "f-c-html-auto-format-toggle")
  end
end
